### ISSUE TITLE
Better i3 Documentation Format

### ISSUE THREAD

**Issue by ghost** [i3/i3#4406]
## I'm submitting a…

<pre>
[ ] Bug
[ ] Feature Request
[x] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

Current documentation; https://i3wm.org/docs/userguide.html

## Desired Behavior

To have something more similar to https://devdocs.io/ much better organized, has sidebar menu and searchable. It also has section with detailed examples. Screenshots are welcome too of course.

## Impact

<pre>
[ ] This feature requires new configuration and/or commands
</pre>
Easier to use for new users and more productive for old users.

## Environment

Output of `i3 --moreversion 2>&-`:
<pre>
i3 version: Any
</pre>

<pre>
- Linux Distribution & Version: Any
- Are you using a compositor (e.g., xcompmgr or compton): No
</pre>

**Comment by Airblader** [comment]
I tend to agree that the documentation format feels dated, and with documentation being important to us – always and even more now that we took a new stance of stability in i3 –, improving the documentation is more than a worthy goal. It'd be a pretty major effort as it requires changing the website infrastructure and probably rewriting big chunks of the documentation.

In any case specific external tooling such as devdocs or whatever would have to be considered first.

**Comment by ghost** [comment]
One of the examples are feature called ``prev_on_output`` and ``next_on_output`` there is no examples for that. But as a newcomer to i3 it took me days to figure this one out. Litterly a week before I tried this commands :-) This is a basic switching between workstations on same monitor, and very similar behavour as I had on XFCE. Documentation was confusing to me as a newbie. After few days I learned to read it well, but I can see how in its current state it can drive users away. I think that having a better documentation will help more new users try and use i3. I'm not saying its bad now, just that it can be much better with more examples ^_^

**Comment by stapelberg** [comment]
We’re already planning moving to markdown in #4327, which we should do first.

Incremental targeted improvements to the documentation are welcome (but please discuss suggestions first before investing too much work).
